pkg/retry/retry.go

```diff
@@ -0,0 +1,80 @@
+package retry
+
+import (
+	"context"
+	"time"
+
+	"github.com/sethvargo/go-retry"
+)
+
+const (
+	defaultInterval    = 1 * time.Second
+	defaultMaxRetries  = 10
+	defaultMaxDuration = 60 * time.Second
+)
+
+// RetryConfig is the configuration for a retry operation.
+type RetryConfig struct {
+	// BackoffStrategy is the backoff strategy to use.
+	BackoffStrategy retry.Backoff
+}
+
+// Retryer is a utility for retrying functions.
+type Retryer struct {
+	config *RetryConfig
+}
+
+// NewNoOpRetryer creates a new Retryer that does not retry.
+func NewNoOpRetryer() *Retryer {
```

Comment:
this is useful for testing. using this retryer should be the same functionality as we have today.